src/app/common/hooks/use-route-header.ts

```diff
@@ -3,10 +3,13 @@ import { useLocation } from 'react-router-dom';
 
 import { useRouteHeaderState } from '@app/store/ui/ui.hooks';
 
-export function useRouteHeader(header: JSX.Element) {
+export function useRouteHeader(header: JSX.Element, isParentRoute?: boolean) {
```

Comment:
@kyranjamie @fbwoolf don't like this solution, but not sure how to do it better

problem is that `useRouteHeader` in parent route component rewrites result of that in child one

Reply:
this function is called on render of child component and then again in parent

Reply:
This is fine I think for now. Updating the header with a hook isn't a great pattern imo.

@mica000 the implementation of the site header we have is really bad and we'd like to replace it. It'd be helpful if we could consider all the variants of header we use throughout the app, so we can reimplement according to that spec.

Reply:
Designs are ready to be implemented on the popover, modal and drawer: https://www.figma.com/file/Li7qK8ZIG9c5dKSNOPO4iCtv/%E2%9D%96-Design-System?node-id=4142%3A6497&t=NY2Wqq9oSogYITWI-4